samples/esb/prx/src/main.c

```diff
@@ -107,27 +105,29 @@ void event_handler(struct esb_evt const *event)
 int clocks_start(void)
 {
 	int err;
-	const struct device *clk;
-
-	clk = device_get_binding(DT_INST_LABEL(0));
-	if (!clk) {
-		LOG_ERR("Clock device not found!");
-		return -EIO;
-	}
+	int res;
+	struct onoff_manager *clk_mgr;
+	struct onoff_client clk_cli;
 
-	err = clock_control_on(clk, CLOCK_CONTROL_NRF_SUBSYS_HF);
-	if (err && (err != -EINPROGRESS)) {
-		LOG_ERR("HF clock start fail: %d", err);
-		return err;
+	clk_mgr = z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF);
+	if (!clk_mgr) {
+		LOG_ERR("Unable to get the Clock manager");
 	}
 
-	/* Block until clock is started.
-	 */
-	while (clock_control_get_status(clk, CLOCK_CONTROL_NRF_SUBSYS_HF) !=
-		CLOCK_CONTROL_STATUS_ON) {
+	sys_notify_init_spinwait(&clk_cli.notify);
 
+	err = onoff_request(clk_mgr, &clk_cli);
+	if (err < 0) {
+		LOG_ERR("Clock request failed: %d", err);
 	}
 
+	do {
+		err = sys_notify_fetch_result(&clk_cli.notify, &res);
+		if (!err && res) {
+			LOG_ERR("Clock could not be started: %d", res);
+		}
```

Comment:
And here, too.

Similarly in `samples/esb/ptx/src/main.c` and `samples/peripheral/radio_test/src/main.c` - those initialization functions should not continue if some their stages ended up with an error.

Reply:
fixed